src/models/geograph.py

```diff
@@ -193,6 +195,14 @@ def crs(self):
         """Return crs of dataframe."""
         return self.df.crs
 
+    def _class_label(self, node_ids: Sequence[int]):
+        """Return class label of `node_ids` directly from underlying numpy array"""
+        return self.df.class_label.values[node_ids]
+
+    def _geometry(self, node_ids: Sequence[int]):
+        """Return geometry of `node_ids` directly from underlying numpy array"""
+        return self.df.geometry.values[node_ids]
+
```

Comment:
What's the anticipated use for these? I'm not sure this really saves code reuse - in fact I feel like most commonly we want to get the full `.values` array. Maybe we could make these a class property so:

```python
@property
def geometry(self):
    return self.df.geometry.values
```
which makes some things a bit neater for calling these outside the class and allows for whatever slicing you want?

Reply:
Great idea acutally! 
The original use was really just to save me typing `self.df.geometry.values` all the time for readibility. But I like your idea much better - will push it with the next commit - thank you for the good suggestion!

Reply:
Done.